hudi-spark-datasource/hudi-spark2/pom.xml

```diff
@@ -199,6 +199,15 @@
       <groupId>org.apache.spark</groupId>
       <artifactId>spark-sql_${scala.binary.version}</artifactId>
       <version>${spark2.version}</version>
+      <scope>provided</scope>
```

Comment:
we also have few other poms that might need similar fixing. 
hudi-spark-datasource/hudi-spark-common
hudi-spark-datasource/hudi-spark3.1.x
hudi-spark-datasource/hudi-spark

Reply:
Good call

Reply:
After closer examination, this `provided` scoping is redundant since all of this packages are marked as "provided" in the root pom